wal/wal.go

```diff
@@ -745,6 +745,9 @@ func (r *Reader) next() (err error) {
 
 		// Gobble up zero bytes.
 		if typ == recPageTerm {
+			// Increase buf to be able to fit an entyre page size minus 1 byte for the rec type.
+			buf = r.buf[1:]
```

Comment:
I have a hard time connecting the dots between this and the linked pull request.
The latter seems to say that we write records of `pageSize` bytes but add the header on top afterwards, which makes the read buffer too small. That's not the case, headers are part of the page, see [here](https://github.com/prometheus/tsdb/blob/master/wal/wal.go#L473).

Could you confirm that this is the actual case the user hit? I'd be curious to understand how it could happen outside of data corruption since we only pad a page with zero bytes (of which `recPageTerm` is the first one), if we can't even fit a header into the page anymore.

Reply:
I have been staring at the code few hours and can't see how can this happen apart from  data corruption.
Will keep digging and will let you know if I find anything else.

Reply:
in any case shouldn't the `buf` be able to fit an entire page after a `recPageTerm` ?

Reply:
Yes, your fix is correct. I'm mostly wondering whether this can happen in the non-corruption case.
I had to go back to the code to understand why we need this though. Could we change the comment to:

```
recPageTerm is a single byte that indicates that the rest of the page is padded. If it's the first byte in a page buf is to small and we have to resize buf to fit pageSize-1 bytes.
```

Reply:
thanks, updated.
Since @AudriusButkevicius won't be available for a while to confirm if this was caused by a data corruption maybe merge this one and will keep his PR open to wait for his reply?

I also opened another PR with to add some more comments if you want to have a look at that one while the topic is fresh.
https://github.com/prometheus/tsdb/pull/430